searx/searx
# Fix google images 'get image' button bug from issue #2103

## What does this PR do?

Fix Google-images engine bug where clicking on 'Get image' button don't give original image URL (issue #2103).

## Why is this change important?

This change is important because it allows Searx users to search full-resolution photos via Google-images. This is very convenient.

## How to test this PR locally?

To test only Google images:
- Go to preferences of your Searx instance.
- Click "Engines" tab and choose "images" section.
- Disable all but google images.
1. Search something like 'wallpaper' and go to images section.
2. Click any image (check if image retrieved with google-images).
3. Press 'Get image' button.
4. See that you have full image URL now instead of page URL.

## Author's checklist

I created `scrap_img_by_id` function with arguments `script` and `data_id` where script is tag selected in line 151. That tag contains all full image URLs. `data_id` is id of image obtained with line 204. It is possible to get image URL with `data_id`, my function do it. Please see [screenshots from my comment](https://github.com/asciimoo/searx/issues/2103#issuecomment-667547267) at issue #2103 for more info.

## Related issues

Closes #2103
